File: auth-center/auth-center-web/src/main/java/com/jmsoftware/maf/authcenter/permission/PermissionRemoteApiController.java
package com.jmsoftware.maf.authcenter.permission;

import com.jmsoftware.maf.authcenter.permission.service.PermissionService;
import com.jmsoftware.maf.common.bean.ResponseBodyBean;
import com.jmsoftware.maf.common.domain.authcenter.permission.GetPermissionListByRoleIdListPayload;
import com.jmsoftware.maf.common.domain.authcenter.permission.GetPermissionListByRoleIdListResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

/**
 * <h1>PermissionRemoteApiController</h1>
 * <p>
 * Change description here.
 *
 * @author Johnny Miller (锺俊), email: johnnysviva@outlook.com
 * @date 5/11/20 8:24 AM
 **/
@RestController
@RequiredArgsConstructor
@RequestMapping("/permission-remote-api")
public class PermissionRemoteApiController {
    private final PermissionService permissionService;

    @GetMapping("/permissions")
    public ResponseBodyBean<GetPermissionListByRoleIdListResponse> getPermissionListByRoleIdList(@Valid GetPermissionListByRoleIdListPayload payload) {
        return ResponseBodyBean.ofSuccess(this.permissionService.getPermissionListByRoleIdList(payload));
    }
}

File: auth-center/auth-center-web/src/main/kotlin/com/jmsoftware/maf/authcenter/permission/PermissionRemoteApiController.kt
package com.jmsoftware.maf.authcenter.permission

import com.jmsoftware.maf.authcenter.permission.service.PermissionService
import com.jmsoftware.maf.common.bean.ResponseBodyBean
import com.jmsoftware.maf.common.domain.authcenter.permission.GetPermissionListByRoleIdListPayload
import com.jmsoftware.maf.common.domain.authcenter.permission.GetPermissionListByRoleIdListResponse
import jakarta.validation.Valid
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * # PermissionRemoteApiController
 *
 * Change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/18/22 8:54 PM
 */
@Validated
@RestController
@RequestMapping("/permission-remote-api")
class PermissionRemoteApiController(
    private val permissionService: PermissionService
) {
    @GetMapping("/permissions")
    fun getPermissionListByRoleIdList(payload: @Valid GetPermissionListByRoleIdListPayload)
            : ResponseBodyBean<GetPermissionListByRoleIdListResponse> =
        ResponseBodyBean.ofSuccess(permissionService.getPermissionListByRoleIdList(payload))
}
